src/rdkafka_fetcher.c

```diff
@@ -650,22 +631,32 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
                 rd_kafka_buf_read_i32(rkbuf, &SessionId);
         }
 
-        rd_kafka_buf_read_i32(rkbuf, &TopicArrayCnt);
```

Comment:
We should use common function `rd_kafka_buf_read_topic_partitions` to read this topic partition.